src/models/db/accounts.js

```diff
@@ -1,36 +1,121 @@
 'use strict'
 
-const assert = require('assert')
 const _ = require('lodash')
-const Account = require('./account').Account
+const assert = require('assert')
 const db = require('../../services/db')
+const knex = require('../../lib/knex').knex
+const config = require('../../services/config')
+
+const TABLE_NAME = 'L_ACCOUNTS'
+
+function convertFromPersistent (data) {
+  data = _.cloneDeep(data)
+  data = _.transform(data, (result, value, key) => {
+    result[key.toLowerCase()] = value
+  })
+  delete data.account_id
+  // some databases store booleans as 0 and 1, and knex does not convert
+  data.is_disabled = Boolean(data.is_disabled)
+  data.is_admin = Boolean(data.is_admin)
+  // oracle has balance stored properly, but knex returns it with a small
+  // rounding error, possibly a bug in knex; using toFixed as workaround
+  data.balance = Number(Number(data.balance).toFixed(config.amount.scale))
+  if (data.minimum_allowed_balance === null) {
+    data.minimum_allowed_balance = Number.NEGATIVE_INFINITY
+  } else if (data.minimum_allowed_balance) {
+    data.minimum_allowed_balance = Number(data.minimum_allowed_balance)
+  } else {
+    data.minimum_allowed_balance = 0
+  }
+  delete data.created_at
+  delete data.updated_at
+  return data
+}
+
+function convertToPersistent (data) {
+  data = _.cloneDeep(data)
+  if (data.balance) {
+    data.balance = Number(Number(data.balance).toFixed(config.amount.scale))
+  }
+  if (data.minimum_allowed_balance) {
+    if (data.minimum_allowed_balance === Number.NEGATIVE_INFINITY) {
+      data.minimum_allowed_balance = null
+    } else {
+      data.minimum_allowed_balance = Number(data.minimum_allowed_balance)
+    }
+  }
+  return _.transform(data, (result, value, key) => {
+    result[key.toUpperCase()] = value
+  })
+}
+
+function getTransaction (options) {
+  return !options ? knex : (!options.transaction ? knex : options.transaction)
+}
 
 function * getAccounts (options) {
-  const jsonAccounts = yield Account.findAll(options)
-  return jsonAccounts.map(Account.fromDatabaseModel.bind(Account))
+  return getTransaction(options)
+    .from(TABLE_NAME).select()
+    .then((results) => results.map(convertFromPersistent))
 }
 
 function * getConnectorAccounts (options) {
-  const jsonAccounts = yield Account.findAll(options)
-  return jsonAccounts.filter((account) => account.connector)
-    .map(Account.fromDatabaseModel.bind(Account))
+  return (yield getAccounts(options))
```

Comment:
Not an issue for this this PR (since this is a refactor), but it might be more efficient if we query the db to return only connector accounts, rather than filtering after we get the response from the db.